controllers/v1/user-extension.js

```diff
@@ -98,6 +98,109 @@ module.exports = class UserExtension extends Abstract {
     })
   }
 
+  /**
+  * @api {get} /kendra/api/v1/user-extension/update/{{userId}} Update user profile
+  * @apiVersion 1.0.0
+  * @apiName Update user profile
+  * @apiGroup User Extension
+  * @apiHeader {String} X-authenticated-user-token Authenticity token
+  * @apiSampleRequest /kendra/api/v1/user-extension/update/e97b5582-471c-4649-8401-3cc4249359bb
+  * @apiUse successBody
+  * @apiUse errorBody
+  * @apiParamExample {json} Response:
+  * {
+  * "message": "User profile fetched successfully.",
+  * "status": 200,
+  *  "result": {
+        "roles": [],
+        "status": "active",
+        "isDeleted": false,
+        "devices": [
+            {
+                "deviceId": "eZPYAJ_thEA:APA91bHniRt_Tfax3KIi0yCqykT-w50-KFyhjgBE9derqXRcFuyu2boa8EVcYn1Yt2lan1eoSNlp2hA5h_oRT7W8YAyYKoQAiT6IuYZ9shJKhDNLnCUR2x1cebrJ3JoMwdIoO5H2Oo7T",
+                "os": "android, android",
+                "app": "diksha survey",
+                "appType": "assessment",
+                "status": "active",
+                "activatedAt": "2020-08-12T10:27:10.888Z"
+            },
+            {
+                "deviceId": "eZPYAJ_thEA:APA91bHniRt_Tfax3KIi0yCqykT-w50-KFyhjgBE9derqXRcFuyu2boa8EVcYn1Yt2lan1eoSNlp2hA5h_oRT7W8YAyYKoQAiT6IuYZ9shJKhDNLnCUR2x1cebrJ3JoMwdIoO5H2Oo7Te",
+                "os": "android, android",
+                "app": "diksha survey",
+                "appType": "assessment",
+                "status": "active",
+                "activatedAt": "2020-08-12T10:28:38.706Z"
+            },
+            {
+                "deviceId": "abc",
+                "os": "android",
+                "app": "diksha-survey",
+                "appType": "assessment",
+                "status": "active",
+                "activatedAt": "2020-08-15T07:51:53.303Z"
+            }
+        ],
+        "userProfileScreenVisitedTrack": null,
+        "deleted": false,
+        "_id": "5f33c3fe39a2e7766b1bef9f",
+        "userId": "01c04166-a65e-4e92-a87b-a9e4194e771d",
+        "externalId": "a1",
+        "createdBy": "SYSTEM",
+        "updatedBy": "SYSTEM",
+        "updatedAt": "2020-09-07T13:24:32.170Z",
+        "createdAt": "2020-08-12T10:27:10.895Z",
+        "improvementProjects": [
+            {
+                "_id": "5f4ce55eb3b81754f08d3528",
+                "name": "Test-2",
+                "description": "improving school library",
+                "externalId": "Test-2",
+                "entityType": "school",
+                "entityTypeId": "5d15a959e9185967a6d5e8a6",
+                "rating": 4
+            }
+        ],
+        "__v": 0
+    }
+  * }
+  */
+
+  /**
+   * Update user profile.
+   * @method
+   * @name update
+   * @param {Object} req - request data.
+   * @param {String} req.params._id - user id.
+   * @returns {JSON} Updated User profile data. 
+   */
+
+  update(req) {
+    return new Promise(async (resolve, reject) => {
+
+      try {
+
+        let result = await userExtensionHelper.update(
```

Comment:
@aman-tunerlabs - We should restrict what can be updated and also just for maybe ratings of project.

Also which is the API for updating or setting roles and entities for API?